src/common/templates/_monitoring.tpl

```diff
@@ -43,3 +43,31 @@ prometheus.io/scrape: {{ $enabled | quote}}
   protocol: "TCP"
 {{- end }}
 {{- end -}}
+
+{{/* Podmonitor template to be added for Google Managed prometheus
+{{- include "harnesscommon.monitoring.podMonitor" (dict "name" "serviceName" ) }}
+*/}}
+{{- define "harnesscommon.monitoring.podMonitor" -}}
+{{- $enabled := and (default (dict) .Values).global.monitoring.enabled (eq ((default (dict) .Values).global.monitoring.managedPlatform) "google") -}}
+{{- $localMonitoring := default (dict) ((pluck "monitoring" .Values) | first) -}}
+{{- $globalMonitoring := default (dict) ((pluck "monitoring" .Values.global) | first) -}}
+{{- $monitoring := (mergeOverwrite $globalMonitoring $localMonitoring ) }}
+{{- $port := (pluck "port" $monitoring) | first }}
+{{- $path := (pluck "path" $monitoring) | first }}
+{{- $namespace := $.Release.Namespace }}
+{{- if $enabled -}}
+apiVersion: monitoring.googleapis.com/v1
+kind: PodMonitoring
+metadata:
+  name: {{ .name }}
+  namespace:  {{ $namespace }}
+spec:
+  selector:
+    matchLabels:
+      app: {{ .name }}
```

Comment:
lets use the labels which are being used in deployment of each service. we can take these labels as arguments as well.